test/functional/readconcern_tests.js

```diff
@@ -1,498 +1,361 @@
-      var configuration = this.configuration;
-      var client = configuration.newClient(
-        { w: 1 },
-        { poolSize: 1, readConcern: { level: 'majority' } }
-      );
-
-      client.connect(function(err, client) {
-        expect(err).to.not.exist;
-
-        var db = client.db(configuration.db);
-        test.deepEqual({ level: 'majority' }, db.s.readConcern);
-
-        // Get a collection
-        var collection = db.collection('readConcernCollection');
-        // Validate readConcern
-        test.deepEqual({ level: 'majority' }, collection.s.readConcern);
-        // Perform a find using the readConcern
-        listener.on('started', function(event) {
-          if (event.commandName === 'find') started.push(event);
-        });
-
-        // Execute find
-        collection.find().toArray(function(err) {
-          expect(err).to.exist;
-
-          listener.uninstrument();
-          client.close();
-          done();
-        });
-      });
+  const insertTests = [
+    {
+      description: 'Should set majority readConcern distinct command',
+      level: 'majority',
+      commandName: 'distinct'
+    },
+    {
+      description: 'Should set majority readConcern count command',
+      level: 'majority',
+      commandName: 'count'
+    },
+    {
+      description: 'Should set majority readConcern group command',
+      level: 'majority',
+      commandName: 'group'
+    },
+    {
+      description: 'Should set majority readConcern parallelCollectionScan command',
+      level: 'majority',
+      commandName: 'parallelCollectionScan'
     }
-  });
-
-  it('Should set majority readConcern aggregate command', {
-    metadata: { requires: { topology: 'replicaset', mongodb: '>= 3.2' } },
-
-    test: function(done) {
-      var listener = require('../..').instrument(function(err) {
-        test.equal(null, err);
-      });
-
-      // Contains all the apm events
-      var started = [];
-      var succeeded = [];
-      // Get a new instance
-      var configuration = this.configuration;
-      var client = configuration.newClient(
-        { w: 1 },
-        { poolSize: 1, readConcern: { level: 'majority' } }
-      );
+  ];
+
+  insertTests.forEach(test => {
+    it(test.description, {
+      metadata: { requires: { topology: 'replicaset', mongodb: '>= 3.2' } },
+
+      test: function(done) {
+        // Get a new instance
+        const configuration = this.configuration;
+        client = configuration.newClient(
+          { w: 1 },
+          { poolSize: 1, readConcern: { level: test.level } }
+        );
 
-      client.connect(function(err, client) {
-        expect(err).to.not.exist;
+        client.connect((err, client) => {
+          expect(err).to.not.exist;
 
-        var db = client.db(configuration.db);
-        test.deepEqual({ level: 'majority' }, db.s.readConcern);
+          const db = client.db(configuration.db);
+          expect({ level: test.level }).to.deep.equal(db.s.readConcern);
```

Comment:
This should be switched to `expect(db.s.readConcern).to.deep.equal({ level: test.level });`. The general pattern is `expect(actual).to.equal(expected)`.